src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersions.java

```diff
@@ -127,6 +128,54 @@ public boolean containsAnyOf(ArtifactVersions versions) {
     return artifactVersions.get().sortByReleaseDate();
   }
 
+  /**
+   * Filter {@link ArtifactVersions} by major version.
+   *
+   * @param version The artifact version.
+   * @return Filtered {@link ArtifactVersions}.
+   */
+  public ArtifactVersions filterArtifactsByMajorVersion(String version) {
+    Optional<SemanticVersion> currentVersion = SemanticVersion.parse(version);
+    if (!currentVersion.isPresent()) {
+      return ArtifactVersions.ofNothing();
+    }
+
+    String currentMajorVersion = buildMajorVersion(currentVersion.get().getMajor());
+    Optional<String> nextMajorVersion = nextMajorVersionOf(version);
+
+    return new ArtifactVersions(elements.stream().filter(v -> v.hasValidSemanticVersion())
```

Comment:
I have a concern with this check
- This will ignore the `ArtifactVersion`s which are non-semantic. Then return the collection for further processing.
- But as a fallback if the returned filtered list contains only single major version `2.0.0`, then that list will contain a non-semantic version.

Please let me know if this is alright? @artem-smotrakov @mibo

Reply:
> But as a fallback if the returned filtered list contains only single major version 2.0.0, then that list will contain a non-semantic version.

Not sure if I understand this sentence.
If the list only contains the `2.0.0` (as the only major version and given as `version` parameter) then the filtered result will contain only this version which is a semantic version.
Why the list should then contain a non-semantic version?

Reply:
@mibo I guess we are on the same page now. Thank you for clearing this up.

Reply:
@sourabhsparkala Sure and Thanks for the nice chat 😉